File: test/test_connection.py
#!/usr/bin/env python3
"""
Simple test script to verify Arrow Flight SQL connection
"""

import pyarrow.flight as flight
import pyarrow as pa

def test_flight_connection():
    try:
        print("🔗 Testing Arrow Flight SQL connection...")
        
        # Create a Flight client
        client = flight.FlightClient("grpc://localhost:32010")
        
        print("✅ Connected to Flight server!")
        
        # Test a simple query
        # Note: This is a basic Flight test, not full SQL
        print("📊 Testing basic Flight operations...")
        
        # List available flights (if any)
        flights = list(client.list_flights())
        print(f"📋 Found {len(flights)} flights")
        
        # Test handshake
        print("🤝 Testing handshake...")
        # Note: Handshake is typically done automatically
        
        print("✅ Basic Flight connection test completed!")
        
    except Exception as e:
        print(f"❌ Connection failed: {e}")
        import traceback
        traceback.print_exc()

if __name__ == "__main__":
    test_flight_connection() 

File: test/TestConnection.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class TestConnection {
    public static void main(String[] args) {
        try {
            System.out.println("🔗 Testing Arrow Flight SQL connection...");
            
            // Load the Arrow Flight SQL driver
            Class.forName("org.apache.arrow.flight.sql.jdbc.FlightSqlDriver");
            
            // Connect to your server
            String url = "jdbc:arrow-flight-sql://localhost:32010";
            Connection conn = DriverManager.getConnection(url);
            
            System.out.println("✅ Connected successfully!");
            
            // Test a simple query
            Statement stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery("SELECT 'Hello from JDBC!' as message");
            
            while (rs.next()) {
                String message = rs.getString("message");
                System.out.println("📊 Query result: " + message);
            }
            
            rs.close();
            stmt.close();
            conn.close();
            
            System.out.println("✅ Test completed successfully!");
            
        } catch (Exception e) {
            System.err.println("❌ Connection failed: " + e.getMessage());
            e.printStackTrace();
        }
    }
} 